src/Hal/Metric/System/Packages/Composer/Packagist.php

```diff
@@ -77,4 +77,51 @@ public function get($package)
 
         return $response;
     }
+
+    /**
+     * Download the given URI and decode it as JSON.
+     *
+     * @param string $uri
+     *
+     * @return mixed
+     */
+    private function getURIContentAsJson($uri)
+    {
+        // Get the environment variable.
+        $httpsProxy = getenv('https_proxy');
+        $context    = null;
+        if ('' !== $httpsProxy) {
+            // Create the context.
+            $context = stream_context_create(
+                [
+                    'http' => [
+                        'proxy' => $this->tcpProtocol($httpsProxy),
```

Comment:
```suggestion
                        'proxy' => str_replace(['http://', 'https://'], 'tcp://', $httpsProxy),
```

And you don't need the `tcpProtocol` method.